### ISSUE TITLE
[video_player] Sometime cannot start play video on iOS

### ISSUE THREAD

**Issue by dodatw** [flutter/flutter#73398]
Hi All,

I am using video_player plug-in. 
I found  a issue that "sometime" it cannot start play video on iOS. (I can reproduce on iphone6s and iphone 7).
The reproduce rate 10%.

I implement a page that auto play a video. 
When I enter page and leave page many times, video freeze at first frame.
Here is sample code that reference video_player sample code. (https://pub.dev/packages/video_player/example)
```
class PlayerVideoAndPopPage extends StatefulWidget {
  @override
  _PlayerVideoAndPopPageState createState() => _PlayerVideoAndPopPageState();
}

class _PlayerVideoAndPopPageState extends State<PlayerVideoAndPopPage> {
  VideoPlayerController _videoPlayerController;
  bool startedPlaying = false;

  @override
  void initState() {
    super.initState();

    _videoPlayerController =
        VideoPlayerController.asset('assets/Butterfly-209.mp4');
    _videoPlayerController.addListener(() {
      if (startedPlaying && !_videoPlayerController.value.isPlaying) {
        Navigator.pop(context);
      }
    });
  }

  @override
  void dispose() {
    _videoPlayerController.dispose();
    super.dispose();
  }

  Future<bool> started() async {
    await _videoPlayerController.initialize();
    await _videoPlayerController.play();
    startedPlaying = true;
    return true;
  }

  @override
  Widget build(BuildContext context) {
    return Material(
      elevation: 0,
      child: GestureDetector(
        onTap: () {
          Navigator.pop(context);
        },
        child: Center(
          child: FutureBuilder<bool>(
            future: started(),
            builder: (BuildContext context, AsyncSnapshot<bool> snapshot) {
              if (snapshot.data == true) {
                return AspectRatio(
                  aspectRatio: _videoPlayerController.value.aspectRatio,
                  child: VideoPlayer(_videoPlayerController),
                );
              } else {
                return const Text('waiting for video to load');
              }
            },
          ),
        ),
      ),
    );
  }
}
```
I am using flutter 1.22.5 and video_player v1.0.1.
Just route to this page and pop. 

No any error can catch when this issue occur. (hasError is false and no catchError found)

**Comment by dodatw** [comment]
Note: the file source is from asset, I can reproduce by sample video Butterfly-209.mp4

**Comment by TahaTesser** [comment]
Hi @dodatw 
Does it reproduce on newer devices like iPhone XR or simulator? 
Can you please provide your `flutter doctor -v` 
Thank you

**Comment by dodatw** [comment]
I just try my iPhone11, I cannot reproduce after enter/leave page 100 times.
On iphone 6s, I can reproduce by enter/leave page <20 times.

flutter doctor -v
```
[✓] Flutter (Channel stable, 1.22.5, on macOS 11.1 20C69 darwin-x64, locale zh-Hant-TW)
    • Flutter version 1.22.5 at /Users/rdapp/git/flutter
    • Framework revision 7891006299 (4 周前), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/rdapp/Library/Android/sdk
    • Platform android-30, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.0

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] Connected device (1 available)
    • Tommy’s iPhone (mobile) • 00008030-0009243A0ED0802E • ios • iOS 14.2

```

**Comment by nt4f04uNd** [comment]
same issue on iphone 7

**Comment by yaodongyi** [comment]
iphonexs也有这个问题

**Comment by darshankawar** [comment]
I verified this issue using vidoe_player plugin [example](https://pub.dev/packages/video_player/example) in iphone 6s device with latest plugin version and stable version with which I didn't observe the reported behavior. The video kept playing properly without freezing upon navigating back and forth.

I will close this issue as not reproducible and may have been resolved after several updates to the plugin since this issue was filed and also the fact there are have been no such reports after last comment. 
If anybody still sees the same behavior with latest plugin version and framework, write in comments and I'll reopen it.
